cc.newspring.Apollos/Rest/Controllers/GiveController.cs

```diff
@@ -206,504 +274,571 @@ public HttpResponseMessage ScheduleGiving( [FromBody]ScheduleParameters schedule
         [System.Web.Http.Route( "api/Give" )]
         public HttpResponseMessage Give( [FromBody]GiveParameters giveParameters )
         {
-            var errorResponse = ValidateGiveParams( giveParameters );
+            var rockContext = new RockContext();
 
-            if ( errorResponse != null )
+            try

```

Comment:
@benwiley I think we talked about this, but wrapping the whole transaction in a try/catch is pretty ugly.  Is this because the context won't return an HTTPResponse inside the transaction?